_rules/error-field-is-not-identified-using-color-only-i3z4v8.md

```diff
@@ -0,0 +1,162 @@
+
+If the presence of errors is conveyed through the use of colors that differ not only in their hue, but that also have a difference in lightness and the difference in relative luminance between the colors leads to a contrast ratio of 3:1 or greater, then this counts as an additional visual mean.
+
+## Assumptions
+
+There are currently no assumptions
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [F81 - Failure of Success Criterion 1.4.1 due to identifying required or error fields using color differences only](https://www.w3.org/WAI/WCAG21/Techniques/failures/F81.html)
+
+It is possible for an error field to be distinguishable from adjacent fields but still not have a meaningful error message. In that case, it would pass Success Criterion 1.4.1: Use of Color but still fail [Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) and, if applicable, [Scuccess Criterion 3.3.3: Error Suggestion](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html).
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The first name `input` value is empty. Its error message (the `p` element) conveyed the presence of an error through its text.
+
+```html
+<label for="first_name">First Name</label>
+<input type="text" id="first_name" autocomplete="given-name" required style="border:1px solid #dd0000" aria-invalid="true" aria-errormessage="first_name_error_msg">
+<p id="first_name_error_msg" class="error_msg" style="color:#dd0000">This field is empty. Enter your first name.</p>
+<label for="last_name" style="color:#333333">Last Name</label>
+<input type="text" id="last_name" autocomplete="family-name" required style="border:1px solid #b0b0b0" value="Doe">
+```
+
+#### Passed Example 2
+
+The phone number `input` value doesn't match the `type` value. Its visual error indicator (the first couple of `label` and `input` elements) conveyed the presence of an error through its CSS properties:
+- font-weight: bold AND;
+- border-width: 3px.
+
+```html
+<label for="phone_number" style="color:#dd0000; font-weight:bold">Phone number</label>
+<input type="tel" id="phone_number" autocomplete="tel" required style="border:3px solid #dd0000" aria-invalid="true" value="John Doe">
+<label for="email" style="color:#333333">Email</label>
+<input type="email" id="email" autocomplete="email" required style="border:1px solid #b0b0b0" value="john.doe@example.com">
+```
+
+#### Passed Example 3
+
+The email `input` value is missing the "@" symbol. Its visual error indicator (the first couple of `label` and `input` elements) conveyed the presence of an error through its CSS properties:
+- lightness (difference in relative luminance between the colors leads to a contrast ratio greater than 3:1) AND;
+- font-weight: bold.
```

Comment:
I was trying to maintain best practices to the passed example. Do you think could be acceptable to remove `font-weight: bold` from the passed example description, but keep it in the passed example code as per new edits?

Reply:
We also need to exert the rule to its limit (and the tools). Since the rule considers that a single difference is enough, we should have test cases with that, even if they are not best practice. This is part of the [One thing at a time](https://act-rules.github.io/pages/design/test-cases/#one-thing-at-a-time) guideline in our design guide.
(actually, the case could even has a contrast ratio of exactly 3:1 to ensure that tools do accept that limit)
We can add to the background that while contrast alone is enough, it is a better practice to also use something else.